### PR TITLE
Send signatures through basket

### PR DESCRIPTION
**Pull request by craigcook**

### CODE REVIEW

File: scripts/email.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    switch (msg) {
+      case errorList.EMAIL_INVALID_ERROR:
+        error = form.querySelector(".error-email-invalid");
+        break;
+      default:
+        error = form.querySelector(".error-try-again-later");
+    }
+
+    if (error) {
+        console.log(error)
```
Comment (robhudson) [inline scripts/email.js:35]: I know this is a draft, but just highlighting this.

File: scripts/form-utils.js [hunk 2]
```diff
@@ -114,22 +103,18 @@ function postToEmailServer(params, successCallback, errorCallback) {
     xhr.timeout = 5000;
     xhr.ontimeout = errorCallback;
     xhr.responseType = 'json';
-
-    if (params.newsletters) {
-      xhr.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
-      xhr.send(serialize(params));
```
Comment (robhudson) [inline scripts/form-utils.js:120]: I set up the basket side more this way -- form urlencoded. The response is JSON though.

File: scripts/form-utils.js [hunk 3]
```diff
@@ -114,22 +103,18 @@ function postToEmailServer(params, successCallback, errorCallback) {
...
 }
 
 function serialize(params) {
+    const name = encodeURIComponent(params.name);
+    const affiliation = encodeURIComponent(params.affiliation);
+    const title = encodeURIComponent(params.title);
     const email = encodeURIComponent(params.email);
-    const newsletters = encodeURIComponent(params.newsletters);
-    const sourceUrl = encodeURIComponent("https://future.mozilla.org");
+    const sourceUrl = encodeURIComponent("https://open.mozilla.org/letter/");
```
Comment (robhudson) [inline scripts/form-utils.js:115]: Should I be doing anything with the source URL?

Comment (craigcook) [reply scripts/form-utils.js:115]: I'm not sure we need it, I can remove

File: scripts/form-utils.js [hunk 4]
```diff
@@ -66,20 +61,14 @@ function enableFormFields(form) {
...
 function postToEmailServer(params, successCallback, errorCallback) {
     const xhr = new XMLHttpRequest();
-    let url = "https://www.mozilla.org/en-US/email-mieco/";
-
-    if (params.newsletters) {
-        url = "https://basket.mozilla.org/news/subscribe/";
-    }
-
-    const { email } = params;
+    let url = "https://basket.allizom.org/petition/sign/";
```
Comment (craigcook) [inline scripts/form-utils.js:64]: So this needs to change to prod before we merge this.

**Review by robhudson — APPROVED** [review]

File: scripts/form-utils.js [hunk 5]
```diff
@@ -114,22 +103,17 @@ function postToEmailServer(params, successCallback, errorCallback) {
     xhr.timeout = 5000;
     xhr.ontimeout = errorCallback;
     xhr.responseType = 'json';
-
```
Comment (robhudson) [inline scripts/form-utils.js:117]: There is no `response.desc` so we can just drop that and call `errorCallback` without a message. Just throw the error if `response.status` is "error".

**Review by robhudson — APPROVED** [review]

**Comment by TExPk5wT7pXAjxdZdkN3Yo5RaGCAv1yDhe** [comment]
1



---Original---
From: "Rob ***@***.***&gt;
Date: Tue, Nov 7, 2023 07:38 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [mozmeao/openletter] Send signatures through basket (PR #12)




 
@robhudson approved this pull request.

**Comment by TExPk5wT7pXAjxdZdkN3Yo5RaGCAv1yDhe** [comment]
@robhudson ```

```Name  =   Value
                                 websocketSR={"data":"close"}

showNote={"data":true}

siteSetting={"data":{"addressSetting":1,"timeSetting":
[true,2]}

USER_UUID={"data":{"uuid":"6511c76b-6d3a-473c-8af9-dd0f351d180a","time":1699355613030}}

language={"data":"zh"}
switch2propertyInfo={"data":true}
ChinaTips={"data":1}
search-session-id={"data":"8c3a3b8c-e26a-408b-85d3-194230c8641d"}
vConsole_switch_y=20
websocket={"data":"close"}
tronHistory={"data":[{"active":false,"value":"TYQ45pHYeRbxFAjBtEvY7xFAUYeKfhYBdU","id":0}]}

islogin={"data":1}
NET={"data":"mainnet"}
vConsole_switch_x=0
websocket-home={"data":"close"}```
``
```

**Pull request merged**